### PR TITLE
Adding a linter to verify Dockerfile

### PR DESCRIPTION
**Pull request by fpesce**
This checks that every dockerfile intending to make an apt-get update
actually performs an upgrade afterwards

This is to prevent any issue similar to #14220

### CODE REVIEW

File: bin/check_dockerfiles.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  while read -r f; do
+    if [ "" != "$(grep "apt-get update" "$f")" ] ; then \
+      if [ "" = "$(grep "apt-get upgrade" "$f")" ] ; then \
+        echo "$f:"; \
+      fi ; \
+    fi; \
+  done)
+
+if [ "" != "$STALE_DOCKERFILES" ]; then
+  exit 1
```
Comment (mandarjog) [inline bin/check_dockerfiles.sh]: `echo $STALE_DOCKERFILE` ?

Comment (mandarjog) [reply bin/check_dockerfiles.sh]: I found `docker/Dockerfile.bionic_debug` was missing this..

Comment (howardjohn) [reply bin/check_dockerfiles.sh]: Yeah looks like the linter failed here but you can't tell why which is a pain when you run into it

Comment (fpesce) [reply bin/check_dockerfiles.sh]: ```docker/Dockerfile.bionic_debug``` is why we had the bug report in the first place.
I'd echo $STALE_DOCKERFILE indeed

**Comment by rlenglet** [comment]
@fpesce Why aren't you using [Hadolint](https://github.com/hadolint/hadolint)?

**Comment by howardjohn** [comment]
@fpesce you need to rebase to pick up the docker changes or tests will fail

**Comment by fpesce** [comment]
> [lint](https://github.com/hadolint/hadolint)?

Because I didn't know abut it :D will have a look,and revisit that PR.

**Pull request closed without merging**